packages/headless/src/api/search/search/product-recommendation.ts

```diff
@@ -2,69 +2,89 @@ export interface ProductRecommendation {
   /**
    * The SKU of the product.
    */
-  sku: string;
+  permanentid: string;
+  /**
+   * A direct link to the product in URL format.
+   */
+  clickUri: string;
   /**
    * Name of the product.
    *
    * From the `ec_name` field.
    */
-  name: string;
-  /**
-   * A direct link to the product in URL format.
-   */
-  link: string;
+  ec_name: string;
```

Comment:
All those changes are breaking changes, and would need to be done with semver philosophy in mind.

Suggestions:

* Keep old fields
* Mark them as deprecated, with an explanation about what should be used instead/their replacement
* Somebody using the "old" field in this version should still have a 100% working product, and would (if possible) get a warning printed in the console on usage about deprecation.
* Create the new fields.
* Keep a mental note/backlog item that when we do the next major version (most likely gonna happen this summer if I have to guess), you will have the opportunity to break/remove deprecated fields.


Or: Live with the consequence and keep the old confusing stuff * forever * (cue evil laugh)

Reply:
Keeping the old fields and adding a `@deprecated` annotation with an explanation of what field to use instead would be best. We could also log a warning: `engine.logger.warn(...)`

Before we release `1.x`, we will go in and remove all the deprecated functionality.